### PR TITLE
Supervisor start finish course

### PR DESCRIPTION
**Pull request by yongsokheng**

### CODE REVIEW

**Comment by hellovietnam93** [comment]
OK

**Comment by NgengChhengkim** [comment]
OK

File: app/models/course.rb [hunk 1]
```diff
@@ -1,7 +1,29 @@
...
+      course_subjects.each do |course_subject|
+        UserSubject.create course_id: id,
+          user_id: user_course.user_id,
+          subject_id: course_subject.subject_id,
+          user_course_id: user_course.id
+      end
+    end
+  end
+
+  def is_progress?
```
Comment (dieunb) [inline app/models/course.rb]: no need

File: app/models/course.rb [hunk 2]
```diff
@@ -1,7 +1,29 @@
...
+
   has_many :course_subjects, dependent: :destroy
   has_many :user_courses, dependent: :destroy
   has_many :user_subjects, dependent: :destroy
   has_many :users, through: :user_courses
   has_many :subjects, through: :course_subjects
+
+  enum status: [:init, :progress, :finish]
+
+  after_update :create_user_subjects, if: :is_progress?
```
Comment (dieunb) [inline app/models/course.rb]: `if: :progress?`

**Pull request merged**